# Hugging Face Streaming Text Load

Stress the agent loop with a streaming text corpus. The default target is `HuggingFaceFW/fineweb-edu` with config `sample-10BT` and a bounded row sample.

Useful knobs: `HF_TEXT_DATASET`, `HF_TEXT_CONFIG`, `HF_TEXT_SPLIT`, `HF_TEXT_FIELD`, `HF_TEXT_ROWS`.

In [1]:
from pathlib import Path

exec(Path("samples/hf_streaming_text_loader.py").read_text(encoding="utf-8"))


dataset: HuggingFaceFW/fineweb-edu
config: sample-10BT
split: train
rows_streamed: 200
mean_chars: 3877.3
max_chars: 65092
top_sources:
- http://austenauthors.net/the-independent-jane: 1
- http://query.nytimes.com/gst/fullpage.html?res=9404E7DA1339F934A25751C0A96E9C8B6: 1
- http://www.childline.org.uk/Explore/SexRelationships/Pages/HIVAIDS.aspx: 1
- http://www.ctt.org/resource_centre/getting_started/learning/understanding_licens: 1
- http://www.environment.ucla.edu/water/news/article.asp?parentid=6178: 1
- http://bayweekly.com/old-site/year99/issue7_40/kids7_40.html: 1
- http://phys.org/news4969.html: 1
- http://www.bbc.co.uk/news/world-latin-america-12292661: 1
top_tokens:
- the: 2639
- and: 1327
- that: 430
- for: 393
- are: 323
- with: 285
- this: 260
- from: 201
- was: 181
- you: 178
- have: 176
- not: 145


row,source,preview
1,http://austenauthors.net/the-independent-jane,"The Independent Jane For all the love, romance and scandal in Jane Austen’s books, what they are really about is freedom and independence. Independence of thought and the freedom to choose. Elizabeth’s refusal of Mr. Collins offer of marria"
2,http://query.nytimes.com/gst/fullpage.html?res=9404E7DA1339F934A25751C0A96E9C8B6,"Taking Play Seriously By ROBIN MARANTZ HENIG Published: February 17, 2008 On a drizzly Tuesday night in late January, 200 people came out to hear a psychiatrist talk rhapsodically about play -- not just the intense, joyous play of children,"
3,http://www.childline.org.uk/Explore/SexRelationships/Pages/HIVAIDS.aspx,"How do you get HIV? HIV can be passed on when infected bodily fluid, such as blood or semen, is passed into an uninfected person. Semen is the liquid which is released from a man's penis during sex which carries sperm. It can be infected wi"
4,http://www.ctt.org/resource_centre/getting_started/learning/understanding_licens,CTComms sends on average 2 million emails monthly on behalf of over 125 different charities and not for profits. Take the complexity of technology and stir in the complexity of the legal system and what do you get? Software licenses! If you
5,http://www.environment.ucla.edu/water/news/article.asp?parentid=6178,Hold the salt: UCLA engineers develop revolutionary new desalination membrane Process uses atmospheric pressure plasma to create filtering 'brush layer' Desalination can become more economical and used as a viable alternate water resource.


## Follow-up Work

After the stream runs, inspect source/domain skew, text length outliers, and whether the sample is representative enough for the experiment. Add a markdown finding before treating the notebook as evidence.

## Finding: HF streaming text clean run

- Result: streamed 200 rows from `HuggingFaceFW/fineweb-edu` config `sample-10BT`, split `train`, with `HF_TEXT_ROWS=200` and the Hugging Face token loaded as `HUGGINGFACE_HUB_TOKEN`.
- Status: no fallback or degradation observed. The run reported `rows_streamed: 200`, `mean_chars: 3877.3`, and `max_chars: 65092`.
- Output shape: source URLs in the reported top sources were all singleton counts at this limit; common-token counts looked plausible for English web text, and the HTML preview rendered representative row snippets.
- Artifacts/cache: no explicit dataset artifact was written by the notebook; the workflow relies on the standard Hugging Face/datasets cache outside the notebook output. The assigned notebook now contains the execution outputs and this finding.
- CLI feedback: `state` made the stable cell id (`hftext01`) easy to target, `run-clean` gave a concise execution status, and `show-cell --outputs full` exposed stdout/stderr plus the HTML preview. The confusing part is that `run-clean` itself does not surface a short output excerpt or the effective environment knobs, so confirming the row limit and dataset requires a follow-up `show-cell`.
